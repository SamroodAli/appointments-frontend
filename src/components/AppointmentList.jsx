import PropTypes from 'prop-types';

const COLOR_ENUM = [
  'indigo',
  'red',
  'green',
  'purple',
  'pink',
  'blue',
];

const AppointmentsList = ({ collection }) => (
  <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4">
    {collection.map(
      (appointment, index) => {
        const color = COLOR_ENUM[index % COLOR_ENUM.length];
        return (
          <div className={`bg-${color}-100 rounded-sm overflow-hidden shadow-lg m-8`} key={appointment.id}>
            <div className="h-32 bg-cover hover:bg-gray" style={{ backgroundImage: 'url(\'https://user-images.githubusercontent.com/5419306/64431443-71823880-d088-11e9-9bd7-1b314f94dc1f.png\')' }} />
            <div className="mx-6 my-4 border-b border-gray-light">
              {/* <div className="font-medium text-base text-gray-darker mb-4">
                  Item name goes here
                  </div> */}
              <p className="font-normal text-gray-dark text-sm mb-2">
                Teacher:
                {' '}
                {appointment.teacher.name}
              </p>
              <p className="font-normal text-gray-dark text-sm mb-4">
                Time:
                {' '}
                {appointment.time}
              </p>
            </div>
          </div>
        );
      },
    )}
  </div>
);

AppointmentsList.propTypes = {
  collection: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};

export default AppointmentsList;
