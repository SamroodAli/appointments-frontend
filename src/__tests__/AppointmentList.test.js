import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
import AppointmentList from '../components/AppointmentList';

test('App snapshot testing', () => {
  const collection = [{ id: 1, teacher: { name: 'teacher1' }, time: 'time1' }, { id: 2, teacher: { name: 'teacher2' }, time: 'time2' }];
  const tree = renderer.create(<AppointmentList collection={collection} />).toJSON();
  expect(tree).toMatchSnapshot();
});
