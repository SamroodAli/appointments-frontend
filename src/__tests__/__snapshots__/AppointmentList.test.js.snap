// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot testing 1`] = `
<div
  className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4"
>
  <div
    className="bg-indigo-100 rounded-sm overflow-hidden shadow-lg m-8"
  >
    <div
      className="h-32 bg-cover hover:bg-gray"
      style={
        Object {
          "backgroundImage": "url('https://user-images.githubusercontent.com/5419306/64431443-71823880-d088-11e9-9bd7-1b314f94dc1f.png')",
        }
      }
    />
    <div
      className="mx-6 my-4 border-b border-gray-light"
    >
      <p
        className="font-normal text-gray-dark text-sm mb-2"
      >
        Teacher:
         
        teacher1
      </p>
      <p
        className="font-normal text-gray-dark text-sm mb-4"
      >
        Time:
         
        time1
      </p>
    </div>
  </div>
  <div
    className="bg-red-100 rounded-sm overflow-hidden shadow-lg m-8"
  >
    <div
      className="h-32 bg-cover hover:bg-gray"
      style={
        Object {
          "backgroundImage": "url('https://user-images.githubusercontent.com/5419306/64431443-71823880-d088-11e9-9bd7-1b314f94dc1f.png')",
        }
      }
    />
    <div
      className="mx-6 my-4 border-b border-gray-light"
    >
      <p
        className="font-normal text-gray-dark text-sm mb-2"
      >
        Teacher:
         
        teacher2
      </p>
      <p
        className="font-normal text-gray-dark text-sm mb-4"
      >
        Time:
         
        time2
      </p>
    </div>
  </div>
</div>
`;
